internal/function_time_parser.go

```diff
@@ -762,36 +762,77 @@ func minuteParser(text []rune, t *time.Time) (int, error) {
 		int(t.Nanosecond()),
 		t.Location(),
 	)
-	return minuteLen, nil
+	return progress, nil
 }
 
 func minuteFormatter(t *time.Time) ([]rune, error) {
 	return []rune(fmt.Sprintf("%02d", t.Minute())), nil
 }
 
-func monthNumberParser(text []rune, t *time.Time) (int, error) {
-	const monthLen = 2
-	if len(text) < monthLen {
-		return 0, fmt.Errorf("unexpected month number")
+func parseDigitRespectingOptionalPlaces(text []rune, minNumber int64, maxNumber int64) (int, int64, error) {
```

Comment:
I'd love a closer look at this function. All test cases seem to show it working well, but I'm not sure how understandable it is! @ageiduschek @colincadams